openshift_day2_configurator/configurators/nodes.py

```diff
@@ -0,0 +1,611 @@
+import shlex
+import shutil
+import os.path
+import logging
+from typing import Any, Dict, Union, Optional
+
+from openshift_day2_configurator.utils.general import (
+    execute_configurator,
+    str_b64encode,
+)
+from rich.progress import Progress
+from timeout_sampler import TimeoutSampler
+from pyhelper_utils.shell import run_command
+from openshift_day2_configurator.utils.resources import create_ocp_resource
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.machine_config import MachineConfig
+from ocp_resources.machine_config_pool import MachineConfigPool
+from ocp_resources.node import Node
+
+MACHINE_CONFIGURATION_ROLE_LABEL: str = "machineconfiguration.openshift.io/role"
+IGNITION_VERSION: str = "3.1.0"
+MASTER_NODE_TYPE: str = "master"
+WORKER_NODE_TYPE: str = "worker"
+NODE_TIMEOUT_7MIN: int = 7 * 60
+
+
+def configure_chrony_ntp_on_nodes_message(nodes_type: str) -> str:
+    return f"Configure Chrony NTP on {nodes_type} nodes"
+
+
+def adding_kernel_arguments_to_nodes_message(nodes_type: str) -> str:
+    return f"Adding kernel arguments to {nodes_type} nodes"
+
+
+def adding_realtime_kernel_to_nodes_message(nodes_type: str) -> str:
+    return f"Adding real-time kernel to {nodes_type} nodes"
+
+
+def configure_journald_setting_on_nodes_message(nodes_type: str) -> str:
+    return f"Configure journald settings on {nodes_type} nodes"
+
+
+def configure_image_registry_setting_on_nodes_message(nodes_type: str) -> str:
+    return f"Configure image registry settings on {nodes_type} nodes"
+
+
+def adding_extensions_to_rhcos_on_nodes_message(nodes_type: str) -> str:
+    return f"Adding extensions to RHCOS on {nodes_type} nodes"
+
+
+def loading_custom_firmware_blobs_on_nodes_message(nodes_type: str) -> str:
+    return f"Loading custom firmware blobs in the machine config manifest for {nodes_type} nodes"
+
+
+def get_machine_config_pool(client: DynamicClient, nodes_type: str) -> MachineConfigPool:
+    return [mcp for mcp in MachineConfigPool.get(dyn_client=client) if mcp.name == nodes_type][0]
+
+
+def get_number_of_nodes(client: DynamicClient, nodes_type: str) -> int:
+    return len([
+        node
+        for node in Node.get(dyn_client=client)
+        if f"node-role.kubernetes.io/{nodes_type}" in node.instance.to_dict().get("metadata", {}).get("labels", {})
```

Comment:
use node.labels

Reply:
save in dict based on node type

Reply:
you mean to create a dict of nodes based on nodes type?
I only need the number of each type

Reply:
ok